From d26fa007d704e7c21f26ec6bce4e2a811ca0f580 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:05:52 +0200
Subject: [PATCH] 1: adapted for saux aarch64

---
 1 | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/1 b/1
index 0bdba54..8f651ee 100755
--- a/1
+++ b/1
@@ -1,11 +1,12 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 . /etc/runit/functions
 
-msg "Welcome to Void!"
+echo
+msg "Welcome to SAUX Aarch64"
 
 [ -r /etc/rc.conf ] && . /etc/rc.conf
 
@@ -28,3 +29,4 @@ install -m000 /dev/null /run/runit/stopit
 install -m000 /dev/null /run/runit/reboot
 
 msg "Initialization complete, running stage 2..."
+echo
-- 
2.28.0

From 178a8502bc8565646c9201d869a5016b6dfacd06 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:08:36 +0200
Subject: [PATCH] 2: adapted for saux-aarch64

---
 2 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/2 b/2
index 0a39298..7f4f2f8 100755
--- a/2
+++ b/2
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 runlevel=default
 for arg in $(cat /proc/cmdline); do
-- 
2.28.0

From 6b4b6d289602b4ea90e2deb7ce24517bcf91998d Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:11:43 +0200
Subject: [PATCH] 3: adapted for saux-aarch64

---
 3 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/3 b/3
index 4a4781e..aaaa5cd 100755
--- a/3
+++ b/3
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 . /etc/runit/functions
 detect_virt
-- 
2.28.0

From 2f512acd4879df89087b95e0054e9e12e791bdd5 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:15:15 +0200
Subject: [PATCH] ctrlaltdel: adapted for saux-aarch64

---
 ctrlaltdel | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ctrlaltdel b/ctrlaltdel
index 866dcc8..4cb0ba9 100755
--- a/ctrlaltdel
+++ b/ctrlaltdel
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 MSG="System is going down..."
 
 # We check for this file after receiving a SIGCONT to move to stage3
-- 
2.28.0

From 976486430cd6331b88602af623ca0025ca2b1293 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:18:17 +0200
Subject: [PATCH] halt.c: adapted for saux-aarch64

---
 halt.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/halt.c b/halt.c
index 3f1062e..3f1c614 100644
--- a/halt.c
+++ b/halt.c
@@ -109,21 +109,21 @@ int main(int argc, char *argv[]) {
     if (do_force)
       reboot(RB_HALT_SYSTEM);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "halt failed");
     break;
   case POWEROFF:
     if (do_force)
       reboot(RB_POWER_OFF);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "poweroff failed");
     break;
   case REBOOT:
     if (do_force)
       reboot(RB_AUTOBOOT);
     else
-      execl("/bin/runit-init", "init", "6", (char*)0);
+      execl("/sbin/runit-init", "init", "6", (char*)0);
     err(1, "reboot failed");
     break;
   case NOOP:
-- 
2.28.0

From 76704547854d04f069907aef53735abe0e3db88d Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:21:03 +0200
Subject: [PATCH] modules-load: adapted for saux-aarch64

---
 modules-load | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules-load b/modules-load
index 35e73b8..2563325 100755
--- a/modules-load
+++ b/modules-load
@@ -1,7 +1,7 @@
 #!/bin/sh
 # modules-load [-n] [-v] - modules-load.d(5) compatible kernel module loader
 
-export PATH=/bin:/sbin
+export PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 {
 # Parameters passed as modules-load= or rd.modules-load= in kernel command line.
-- 
2.28.0

From 82bf341621960d4cb53268972ecaf563204f0d5b Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:25:04 +0200
Subject: [PATCH] rc.conf: adapted for saux-aarch64

---
 rc.conf | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/rc.conf b/rc.conf
index 5dbad2e..7dace51 100644
--- a/rc.conf
+++ b/rc.conf
@@ -5,19 +5,19 @@
 # NOTE: it's preferred to declare the hostname in /etc/hostname instead:
 # 	- echo myhost > /etc/hostname
 #
-#HOSTNAME="void-live"
+HOSTNAME="host"
 
 # Set RTC to UTC or localtime.
-#HARDWARECLOCK="UTC"
+HARDWARECLOCK="UTC"
 
 # Set timezone, availables timezones at /usr/share/zoneinfo.
-#TIMEZONE="Europe/Madrid"
+TIMEZONE="UTC"
 
 # Keymap to load, see loadkeys(8).
-#KEYMAP="es"
+KEYMAP="us"
 
 # Console font to load, see setfont(8).
-#FONT="lat9w-16"
+FONT="default"
 
 # Console map to load, see setfont(8).
 #FONT_MAP=
-- 
2.28.0

From 8f7b00299d27eff56bada46e83013edc3682ac76 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:35:35 +0200
Subject: [PATCH] Makefile: adapted for saux-aarch64

---
 Makefile | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 877969b..c583aee 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
-PREFIX ?=	/usr/local
+PREFIX ?=	/
+MANDIR =	/usr/share/man
 SCRIPTS=	1 2 3 ctrlaltdel
 
 all:
@@ -17,16 +18,16 @@ install:
 	ln -sf zzz ${DESTDIR}/${PREFIX}/sbin/ZZZ
 	ln -sf halt ${DESTDIR}/${PREFIX}/sbin/poweroff
 	ln -sf halt ${DESTDIR}/${PREFIX}/sbin/reboot
-	install -d ${DESTDIR}/${PREFIX}/share/man/man1
-	install -m644 pause.1 ${DESTDIR}/${PREFIX}/share/man/man1
-	install -d ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 zzz.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 shutdown.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 halt.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 modules-load.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 vlogger.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	ln -sf halt.8 ${DESTDIR}/${PREFIX}/share/man/man8/poweroff.8
-	ln -sf halt.8 ${DESTDIR}/${PREFIX}/share/man/man8/reboot.8
+	install -d ${DESTDIR}/${PREFIX}/${MANDIR}/man1
+	install -m644 pause.1 ${DESTDIR}/${PREFIX}/${MANDIR}/man1
+	install -d ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 zzz.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 shutdown.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 modules-load.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 vlogger.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	ln -sf halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8/poweroff.8
+	ln -sf halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8/reboot.8
 	install -d ${DESTDIR}/etc/sv
 	install -d ${DESTDIR}/etc/runit/runsvdir
 	install -d ${DESTDIR}/etc/runit/core-services
-- 
2.28.0

From c9bb3fb379668af96b05884f127f98a97e0ebbe7 Mon Sep 17 00:00:00 2001
From: sajcho <saux.aarch64@gmail.com>
Date: Thu, 15 Oct 2020 15:42:20 +0200
Subject: [PATCH] 99-cleanup.sh: adapted for saux-aarch64

---
 core-services/99-cleanup.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core-services/99-cleanup.sh b/core-services/99-cleanup.sh
index f27f728..57de2c1 100644
--- a/core-services/99-cleanup.sh
+++ b/core-services/99-cleanup.sh
@@ -1,4 +1,5 @@
 # vim: set ts=4 sw=4 et:
+cd /tmp && /usr/bin/find . ! -name . -delete
 
 if [ ! -e /var/log/wtmp ]; then
 	install -m0664 -o root -g utmp /dev/null /var/log/wtmp
-- 
2.28.0

