From 3c89f022a00fc7cc476179efe669ee2e893b2af2 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:28:49 +0200
Subject: [PATCH 01/11] Makefile: chenge build and install path

---
 Makefile | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 877969b..c583aee 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
-PREFIX ?=	/usr/local
+PREFIX ?=	/
+MANDIR =	/usr/share/man
 SCRIPTS=	1 2 3 ctrlaltdel
 
 all:
@@ -17,16 +18,16 @@ install:
 	ln -sf zzz ${DESTDIR}/${PREFIX}/sbin/ZZZ
 	ln -sf halt ${DESTDIR}/${PREFIX}/sbin/poweroff
 	ln -sf halt ${DESTDIR}/${PREFIX}/sbin/reboot
-	install -d ${DESTDIR}/${PREFIX}/share/man/man1
-	install -m644 pause.1 ${DESTDIR}/${PREFIX}/share/man/man1
-	install -d ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 zzz.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 shutdown.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 halt.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 modules-load.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	install -m644 vlogger.8 ${DESTDIR}/${PREFIX}/share/man/man8
-	ln -sf halt.8 ${DESTDIR}/${PREFIX}/share/man/man8/poweroff.8
-	ln -sf halt.8 ${DESTDIR}/${PREFIX}/share/man/man8/reboot.8
+	install -d ${DESTDIR}/${PREFIX}/${MANDIR}/man1
+	install -m644 pause.1 ${DESTDIR}/${PREFIX}/${MANDIR}/man1
+	install -d ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 zzz.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 shutdown.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 modules-load.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	install -m644 vlogger.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8
+	ln -sf halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8/poweroff.8
+	ln -sf halt.8 ${DESTDIR}/${PREFIX}/${MANDIR}/man8/reboot.8
 	install -d ${DESTDIR}/etc/sv
 	install -d ${DESTDIR}/etc/runit/runsvdir
 	install -d ${DESTDIR}/etc/runit/core-services
-- 
2.27.0

From db78c01e2987b88bdfcce89c3bf564fb68856fe0 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:37:24 +0200
Subject: [PATCH 02/11] file 1: added to PATH /bin, /sbin

---
 1 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/1 b/1
index 6938bc8..5164eba 100755
--- a/1
+++ b/1
@@ -1,11 +1,11 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 . /etc/runit/functions
 
-msg "Welcome to Void!"
+msg "Welcome to SAUX Aarch64!"
 
 [ -r /etc/rc.conf ] && . /etc/rc.conf
 
-- 
2.27.0

From 29c83a24862d942fa386235b450ff124bbd4fca5 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:43:08 +0200
Subject: [PATCH 03/11] file 2: added to PATH /bin, /sbin

---
 2 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/2 b/2
index 0a39298..7f4f2f8 100755
--- a/2
+++ b/2
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 runlevel=default
 for arg in $(cat /proc/cmdline); do
-- 
2.27.0

From eb4ad26fc5c720a193158550174eb7a22a5b0295 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:44:51 +0200
Subject: [PATCH 04/11] file 3: added to PATH /bin, /sbin

---
 3 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/3 b/3
index 7945490..4416763 100755
--- a/3
+++ b/3
@@ -1,7 +1,7 @@
 #!/bin/sh
 # vim: set ts=4 sw=4 et:
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 . /etc/runit/functions
 detect_virt
-- 
2.27.0

From 53832059c8cf646c230ef103ddf3ace1cd5ed1b6 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:48:25 +0200
Subject: [PATCH 05/11] ctrlaltdel: added to PATH /bin, /sbin

---
 ctrlaltdel | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ctrlaltdel b/ctrlaltdel
index 7738067..847bee9 100755
--- a/ctrlaltdel
+++ b/ctrlaltdel
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-PATH=/usr/bin:/usr/sbin
+PATH=/bin:/usr/bin:/sbin:/usr/sbin
 MSG="System is going down..."
 
 # We check for this file in stage3 to halt or reboot
-- 
2.27.0

From 7ff30926774fcc8d5170b0f617e7dd9903bfe38e Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:54:06 +0200
Subject: [PATCH 06/11] halt.c: change execl path - /bin on /sbin

---
 halt.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/halt.c b/halt.c
index 3f1062e..3f1c614 100644
--- a/halt.c
+++ b/halt.c
@@ -109,21 +109,21 @@ int main(int argc, char *argv[]) {
     if (do_force)
       reboot(RB_HALT_SYSTEM);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "halt failed");
     break;
   case POWEROFF:
     if (do_force)
       reboot(RB_POWER_OFF);
     else
-      execl("/bin/runit-init", "init", "0", (char*)0);
+      execl("/sbin/runit-init", "init", "0", (char*)0);
     err(1, "poweroff failed");
     break;
   case REBOOT:
     if (do_force)
       reboot(RB_AUTOBOOT);
     else
-      execl("/bin/runit-init", "init", "6", (char*)0);
+      execl("/sbin/runit-init", "init", "6", (char*)0);
     err(1, "reboot failed");
     break;
   case NOOP:
-- 
2.27.0

From 3dc052d2160b432654a69a3150c85281a770e89c Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 16:59:38 +0200
Subject: [PATCH 07/11] modules-load: added to PATH /usr/bin, /usr/sbin

---
 modules-load | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules-load b/modules-load
index 35e73b8..2563325 100755
--- a/modules-load
+++ b/modules-load
@@ -1,7 +1,7 @@
 #!/bin/sh
 # modules-load [-n] [-v] - modules-load.d(5) compatible kernel module loader
 
-export PATH=/bin:/sbin
+export PATH=/bin:/usr/bin:/sbin:/usr/sbin
 
 {
 # Parameters passed as modules-load= or rd.modules-load= in kernel command line.
-- 
2.27.0

From 2a5c0729566c4549e747fb9f65520faa3586a003 Mon Sep 17 00:00:00 2001
From: Milan Buska <milan.buska@gmail.com>
Date: Fri, 26 Jun 2020 17:08:09 +0200
Subject: [PATCH 08/11] rc.conf: change variable HOSTNAME, TIMEZONE, KEYMAP,
 FONT

---
 rc.conf | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/rc.conf b/rc.conf
index 5dbad2e..7dace51 100644
--- a/rc.conf
+++ b/rc.conf
@@ -5,19 +5,19 @@
 # NOTE: it's preferred to declare the hostname in /etc/hostname instead:
 # 	- echo myhost > /etc/hostname
 #
-#HOSTNAME="void-live"
+HOSTNAME="host"
 
 # Set RTC to UTC or localtime.
-#HARDWARECLOCK="UTC"
+HARDWARECLOCK="UTC"
 
 # Set timezone, availables timezones at /usr/share/zoneinfo.
-#TIMEZONE="Europe/Madrid"
+TIMEZONE="UTC"
 
 # Keymap to load, see loadkeys(8).
-#KEYMAP="es"
+KEYMAP="us"
 
 # Console font to load, see setfont(8).
-#FONT="lat9w-16"
+FONT="default"
 
 # Console map to load, see setfont(8).
 #FONT_MAP=
-- 
2.27.0

